src/app/auth/auth.service.ts

```diff
@@ -1,237 +1,180 @@
-import firebase from 'firebase/compat/app';
 import { none } from 'fp-ts/lib/Option';
-import { Observable, Subscription, from, of } from 'rxjs';
-import { map, switchAll, switchMap, take, tap } from 'rxjs/operators';
+import { Observable, from } from 'rxjs';
+import { map } from 'rxjs/operators';
 import { BaseService } from '../core/base.service';
 import { AlertService, Level, UntranslatedAlertMessage } from '../messaging/alert.service';
-import { User } from '../profile/user';
-import { FirestoreDebugService } from '../shared/firestore-debug.service';
 import { LocalService } from '../shared/local.service';
-import { LoginResult } from './login-result';
 
-export const LOGIN_URL = '/auth/login';
-export const LOGGED_OUT_URL = '/auth/logged-out';
-
-const LOCALSTORAGE_LOGIN_RESULT_KEY = 'loginResult';
+const LOCALSTORAGE_LOGGEDIN_KEY = 'loggedin';
 
 @Injectable()
-export class AuthService extends BaseService implements OnDestroy {
+export class AuthService extends BaseService {
   browserIdHeaders: HttpHeaders;
+  private afAuth = inject(Auth);
+
+  /**
+   * @deprecated use signals instead
+   */
+  public firebaseUser$ = authState(this.afAuth);
 
-  private subscriptions = new Subscription();
-  public user$: Observable<User>;
-  public firebaseUser$: Observable<firebase.User>;
+  public authenticated: Signal<boolean>;
+  public uid: Signal<string | null>;
+  public email: Signal<string | null>;
 
-  // store the URL so we can redirect after logging in
-  redirectUrl: string;
+  redirectUrl: string; // TODO move this -> login components
 
   constructor(
     alertService: AlertService,
-    private router: Router,
-    private afAuth: AngularFireAuth,
     private afs: AngularFirestore,
-    private fds: FirestoreDebugService,
     private localService: LocalService
   ) {
     super(alertService);
-
-    this.firebaseUser$ = this.afAuth.authState.pipe(
-      // multiple events on login
-      tap(() => this.fds.addRead('firebaseUser'))
+    const firebaseUser$ = authState(this.afAuth);
+    const firebaseUser = toSignal(firebaseUser$);
+
+    this.authenticated = toSignal(
+      firebaseUser$.pipe(
+        map(user => !!user) // Convert user object to boolean
+      ),
+      { initialValue: this.isCachedLoggedIn() }
     );
-    this.user$ = this.firebaseUser$.pipe(
-      switchMap(user => {
-        if (user) {
-          return this.afs
-            .doc<User>(`users/${user.uid}`)
-            .valueChanges()
-            .pipe(tap(() => this.fds.addRead('users')));
-        } else {
-          this.resetClientSession();
-          return of(null) as Observable<User>;
-        }
-      })
-    );
-    this.subscriptions.add(this.user$.subscribe());
-  }
 
-  ngOnDestroy(): void {
-    this.subscriptions.unsubscribe();
+    this.uid = computed(() => firebaseUser()?.uid);
+    /**
+     * // TODO This signal is not updated on email changes!
+     */
+    this.email = computed(() => firebaseUser()?.email);
```

Comment:
email has to update on email change, causing problems eg. in profile edit. The change is not triggered by the authState.